.github/workflows/test.yml

```diff
@@ -37,38 +37,6 @@ jobs:
         with:
           node-version: ${{ matrix.node }}
 
-      - uses: pnpm/action-setup@v2
-        name: Install pnpm
-        id: pnpm-install
-        with:
-          version: 8
-          run_install: false
-
-      - name: Get pnpm store directory
-        id: pnpm-cache
-        shell: bash
-        run: |
-          echo "STORE_PATH=$(pnpm store path)" >> $GITHUB_OUTPUT
-
-      - name: Cache node modules
-        id: cache-node-modules
-        uses: actions/cache@v4
-        env:
-          cache-name: cache-node-modules
-        with:
-          path: ${{ steps.pnpm-cache.outputs.STORE_PATH }}
-          key: ${{ runner.os }}-pnpm-store-${{ env.cache-name }}-${{ hashFiles('**/pnpm-lock.yaml') }}
-          restore-keys: |
-            v1-${{ runner.os }}-pnpm-store-${{ env.cache-name }}-
-            v1-${{ runner.os }}-pnpm-store-
-            v1-${{ runner.os }}-
-
-      - name: Install project dependencies
-        run: pnpm install
-
-      - name: Build CLI
```

Comment:
this was probably just leftover cruft, and there was no point in running build during the install step, so moved building to right before running the tests below

Reply:
actually I misunderstood this, so the way this is working is it's running the build step in install job, so it builds for `node18` and `node20` then it caches the build and passes it to the test job which runs for a mix of `node + shard matrix` if the `build` step is still needed we should keep the cache so it doesn't need to build on each matrix which would take quite long for 4 shards if cache is not available.

I thought the build step isn't required at all but maybe that will not work. Happy to chat more on this

Reply:
> I thought the build step isn't required at all but maybe that will not work. Happy to chat more on this

You're right! Jest is configured to use the dev aliases, so tests can run from source. Although discovered that the validation tests, requires these aliases to be registered inside the worker thread that it runs.

Opened a PR to fix this here: https://github.com/sanity-io/sanity/pull/5927